baseplate/observers/metrics_tagged.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import typing
+
+from typing import Any
+from typing import Optional
+
+from baseplate import _ExcInfo
+from baseplate import BaseplateObserver
+from baseplate import LocalSpan
+from baseplate import RequestContext
+from baseplate import Span
+from baseplate import SpanObserver
+from baseplate.lib import config
+from baseplate.lib import metrics
+from baseplate.observers.timeout import ServerTimeout
+
+
+logger = logging.getLogger(__name__)
+
+
+class TaggedMetricsBaseplateObserver(BaseplateObserver):
+    """Metrics collecting observer.
+
+    This observer reports metrics to statsd in the dogstatsd format. It does three important things:
+
+    * it tracks the time taken in serving each request.
+    * it batches all metrics generated during a request into as few packets
+      as possible.
+    * it adds tags to the metric if they are in the config tag whitelist
+
+    The batch is accessible to your application during requests as the
+    ``metrics`` attribute on the :py:class:`~baseplate.RequestContext`.
+
+    :param client: The client where metrics will be sent.
+    :param cfg: the parsed application config with the tag whitelist
+
+    """
+
+    def __init__(self, client: metrics.Client, cfg: config.RawConfig):
+        self.client = client
+        self.whitelist = self.parse_tags(cfg)
+
+    def on_server_span_created(self, context: RequestContext, server_span: Span) -> None:
+        batch = self.client.tagged_batch()
+        context.metrics = batch
+        observer = TaggedMetricsServerSpanObserver(batch, server_span, self.whitelist)
+        server_span.register(observer)
+
+    def parse_tags(self, raw_config: config.RawConfig) -> list:
+        cfg = config.parse_config(
+            raw_config,
+            {
+                "metrics": {
+                    "tagging": config.Optional(config.Boolean, default=False),
+                    "whitelist": config.Optional(config.String, default=""),
```

Comment:
does this mean that every application will need to add the built-in tags to the whitelist? maybe we should just hard-code those as acceptable?

Reply:
Currently that's how it is set up because I was concerned about the cardinality limits (especially the 20 tag hard cap). Open to hard coding them if that's not a concern though

Reply:
just add stuff in default= 
* endpoint
* success